Emby.Server.Implementations/HttpServer/WebSocketConnection.cs

```diff
@@ -231,6 +231,15 @@ private Task SendKeepAliveResponse()
                 CancellationToken.None);
         }
 
+        /// <summary>
+        /// Gracefully closes the socket.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        public void CloseSocket(CancellationToken cancellationToken)
+        {
+            _socket.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, "System Shutdown", cancellationToken);
```

Comment:
This needs an `await`

Reply:
Done, I also made everything in the call stack async too